base/src/main/java/io/spine/reflect/PackageGraph.java

```diff
@@ -116,39 +115,36 @@ public static PackageGraph containing(String packagePrefix) {
      */
     public static PackageGraph matching(Predicate<Package> predicate) {
         checkNotNull(predicate);
-        List<Package> filteredPackages = filterPackages(predicate);
+        var filteredPackages = filterPackages(predicate);
         return create(filteredPackages);
     }
 
     private static List<Package> filterPackages(Predicate<Package> predicate) {
-        List<Package> result = packages().stream()
-                                         .filter(predicate)
-                                         .collect(toList());
+        var result = packages().stream()
+                .filter(predicate)
+                .collect(toList());
         return result;
     }
 
     private static PackageGraph create(List<Package> filteredPackages) {
-        Graph<PackageInfo> mutableGraph = buildGraph(filteredPackages);
-        PackageGraph result = new PackageGraph(mutableGraph);
+        var mutableGraph = buildGraph(filteredPackages);
+        var result = new PackageGraph(mutableGraph);
         return result;
     }
 
     private static Graph<PackageInfo> buildGraph(List<Package> packages) {
-        MutableGraph<PackageInfo> graph = GraphBuilder
-                .directed()
-                .nodeOrder(ElementOrder.<PackageInfo>natural())
-                .build();
+        var graph = GraphBuilder.directed()
+                                .nodeOrder(ElementOrder.<PackageInfo>natural())
+                                .build();
         Queue<Package> queue = new ArrayDeque<>(packages);
-        Package first = queue.poll();
+        var first = queue.poll();
         while (first != null) {
-            final Package current = first;
-            Optional<PackageInfo> directParent =
-                    graph.nodes()
-                         .stream()
-                         .filter((node) -> IsDirectParent.of(current)
-                                                         .test(node.getValue()))
-                         .findFirst();
-            PackageInfo newNode = PackageInfo.of(current);
+            final var current = first;
+            var directParent = graph.nodes().stream()
+                    .filter((node) -> IsDirectParent.of(current)
```

Comment:
Please extract `IsDirectParent.of(current)` from the lambda. Also, there is something strange going on with the renaming `first` to `current`. Maybe, we could get rid of that as well.